lib/ansible/modules/network/ftd/ftd_install.py

```diff
@@ -0,0 +1,305 @@
+      - Defaulted to 'cisco.com' if not specified.
+    required: false
+    type: str
+    default: cisco.com
+"""
+
+EXAMPLES = """
+  - name: Install image v6.3.0 on FTD 5516
+    ftd_install:
+      device_hostname: firepower
+      device_password: pass
+      device_ip: 192.168.0.1
+      device_netmask: 255.255.255.0
+      device_gateway: 192.168.0.254
+      dns_server: 8.8.8.8
+
+      console_ip: 10.89.0.0
+      console_port: 2004
+      console_username: console_user
+      console_password: console_pass
+
+      rommon_file_location: 'tftp://10.89.0.11/installers/ftd-boot-9.10.1.3.lfbff'
+      image_file_location: 'https://10.89.0.11/installers/ftd-6.3.0-83.pkg'
+      image_version: 6.3.0-83
+"""
+
+RETURN = """
+msg:
+    description: The message saying whether the image was installed or explaining why the installation failed.
+    returned: always
+    type: str
+"""
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.connection import Connection
+from ansible.module_utils.six import iteritems
+
+from ansible.module_utils.network.ftd.configuration import BaseConfigurationResource, ParamName, \
+    PATH_PARAMS_FOR_DEFAULT_OBJ
+from ansible.module_utils.network.ftd.device import HAS_KICK, FtdPlatformFactory, FtdModel
+
+REQUIRED_PARAMS_FOR_LOCAL_CONNECTION = ['device_ip', 'device_netmask', 'device_gateway', 'device_model', 'dns_server']
+
+
+class FtdOperations:
+    GET_SYSTEM_INFO = 'getSystemInformation'
+    GET_MANAGEMENT_IP_LIST = 'getManagementIPList'
+    GET_DNS_SETTING_LIST = 'getDeviceDNSSettingsList'
+    GET_DNS_SERVER_GROUP = 'getDNSServerGroup'
+
+
+def main():
+    fields = dict(
+        device_hostname=dict(type='str', required=True),
+        device_username=dict(type='str', required=False, default='admin'),
+        device_password=dict(type='str', required=True, no_log=True),
+        device_sudo_password=dict(type='str', required=False, no_log=True),
+        device_new_password=dict(type='str', required=False, no_log=True),
+        device_ip=dict(type='str', required=False),
+        device_netmask=dict(type='str', required=False),
+        device_gateway=dict(type='str', required=False),
+        device_model=dict(type='str', required=False, choices=FtdModel.supported_models()),
+        dns_server=dict(type='str', required=False),
+        search_domains=dict(type='str', required=False, default='cisco.com'),
+
+        console_ip=dict(type='str', required=True),
+        console_port=dict(type='str', required=True),
+        console_username=dict(type='str', required=True),
+        console_password=dict(type='str', required=True, no_log=True),
+
+        rommon_file_location=dict(type='str', required=True),
+        image_file_location=dict(type='str', required=True),
+        image_version=dict(type='str', required=True),
+        force_install=dict(type='bool', required=False, default=False)
+    )
+    module = AnsibleModule(argument_spec=fields)
+    if not HAS_KICK:
```

Comment:
This exception can be handled in module_utils code itself from __init__ method:
```suggestion
    if not HAS_KICK:
       raise Exception('Kick Python module is required to run this module. Please, install it with `pip install kick` command and run the playbook again.')
```

Reply:
Do you mean I should rather throw an exception explicitly instead of calling `module.fail_json`? I just checked other modules (e.g., [github_release](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/github_release.py#L167) or [gcp](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/gcp.py#L339)) and followed the same approach.

Reply:
@annikulin No, from the above comment I meant was that you shall handle respective scenario in `module_utils` code rather than `module` code, this way if the same functionality is needed by any future ftd resource module code won't be duplicated (e.g.: [nios_api](https://github.com/ansible/ansible/blob/5feadbef4a75b38b450820c9c5cc6e55e71e65fe/lib/ansible/module_utils/net_tools/nios/api.py#L36), [panos](https://github.com/ansible/ansible/blob/5feadbef4a75b38b450820c9c5cc6e55e71e65fe/lib/ansible/module_utils/network/panos/panos.py#L34)).

Reply:
@justjais, ok, that makes sense. I moved the part checking that the library is installed to `module_utils`. I also passed a `module` to this method, so that we invoke `fail_json` and get a short and descriptive error message instead of raising an exception and handling it in every module.